### PR TITLE
Create Animal Facts.js

### PR DESCRIPTION
**Pull request by sdgrube2**
Animal Facts parser for slackerbot. !animal (animal name) returns various results of features and characteristics of the top 3 related animals.

### CODE REVIEW

File: Parsers/Animal Facts.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+regex:!animal
+flags:gmi
+*/
+
+var where = current.text.indexOf('!animal ') + 8;
+var term = current.text.substr(where).trim();
+var chatReq = new sn_ws.RESTMessageV2();
+chatReq.setEndpoint('https://api.api-ninjas.com/v1/animals?name=' + term);
+chatReq.setHttpMethod("GET");
+chatReq.setRequestHeader('X-Api-Key', 'tbfgkwW8c0UY0xE0Yd9oSw==DusJx995ZmqecaPW');
```
Comment (SapphicFire) [inline Parsers/Animal Facts.js:12]: Merging would share your key more broadly. I'd suggest putting this into a global property and then requesting that it be added by the team

File: Parsers/Animal Facts.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+var chatResponse = chatReq.execute();
+var chatResponseBody = chatResponse.getBody();
```
Comment (SapphicFire) [inline Parsers/Animal Facts.js:14]: I would suggest adding error handling so that the parser can fail gracefully if it encounters issues

**Review by SapphicFire — CHANGES_REQUESTED** [review]

**Review by sdgrube2 — COMMENTED** [review]
Removed key and created a system property.
I will commit change and submit PR separately for that system property.
If i'm doing that wrong please let me know, i'm very junior.
I changed this parser to now shuffle animals for semi-random results instead of just top 3 each time.
I added code to fail gracefully if results are out of bounds of what is requested from the server.

**Comment by earlduque** [comment]
Hi again, as suggested by @SapphicFire, we try not to include extra properties into the repository so that end-users that choose to use this repository can choose which parsers (and their relevant properties) they want in their instances. Will you bring this back to just a parser and then message one of the reviewers on sndevs, community, or linkedin the property name and value so that we can make it a global property. and then we will create the property for you on the instance this is running on and update documentation to note the needed property if people are installing this at home

**Comment by sdgrube2** [comment]
@earlduque Sorry -- Please let me know if I'm missing something obvious here. This should just be the parser with the key removed and replaced with a name of a system property I made up.  I can then message the approvers the name of the required system property and the corresponding key so then it can work.  Do I need to close this and just do another PR with the new script? That should be what this is now, but maybe the history of it keeps the old script with the key? Sorry, just confused. I really appreciate the help!

**Comment by SapphicFire** [comment]
Hi @sdgrube2 - it's a little nuanced. When I say a 'Global' property, I mean one that is done under that scope. The property created against this one was under the SlackerBot scope (`x_snc_slackerbot.SlackerBot.animal.token`), which means it is included in the scope with the token, and additionally for any commits by other users. The solution would be to delete that property, change your scope to Global, then create it again. 

You'd also want to update your `.getProperty('value','')` call to look for the new property, and send a message that the property wasn't set if someone tried to use the parser.

I appreciate your patience, and apologise for being unclear when I said global property.

**Review by sdgrube2 — COMMENTED** [review]
Changed parser as requested.
I removed the API key, revoked it since it went public, and a new one is referenced as a global property in the script as 'slackerbot.animal.token' which I will submit with key value to admins.
Hopefully this script is much better. Works great on my slack channel. Thanks for the help everyone! Please let me know if there are other suggestions or changes to be made!

**Review by SapphicFire — APPROVED** [review]

**Review by SapphicFire — CHANGES_REQUESTED** [review]
Hi @sdgrube2, awesome work with the parser. One last thing, please remove the application files (`b02cf9e61b861d10d806dc23b24bcb3f` and its contents), because these are not within scope of your update. Please reach out if you need a hand in how to do that.

**Review by sdgrube2 — COMMENTED** [review]
Deleted unnecessary directory

**Comment by sdgrube2** [comment]
Global property created. Ready for merge, if acceptable.

**Comment by SapphicFire** [comment]
Hi @sdgrube2 - thanks for making that change. However, by deleting it did unfortunately remove the entire application - I apologise as I should have been clearer. As I know this has taken a while and didn't want to drag it out for you, I reverted to a clean commit of yours and then re-committed your final version. As we're all good to go, I'll go ahead and accept this. Thank you for your work and patience.

**Review by SapphicFire — APPROVED** [review]
Looks great to me, thank you!

**Comment by sdgrube2** [comment]
> Hi @sdgrube2 - thanks for making that change. However, by deleting it did unfortunately remove the entire application - I apologise as I should have been clearer. As I know this has taken a while and didn't want to drag it out for you, I reverted to a clean commit of yours and then re-committed your final version. As we're all good to go, I'll go ahead and accept this. Thank you for your work and patience.

Thank you so much! I really appreciate the help with this, and sorry for making it more difficult than it needed to be. I’ll learn :)

**Pull request merged**